rhino-core/src/main/java/io/ryos/rhino/sdk/runners/ReactiveHttpSimulationRunner.java

```diff
@@ -101,30 +106,63 @@ public void start() {
     this.subscribe = Flux.fromStream(Stream.generate(userRepository::take))
         .take(simulation.getDuration())
         .zipWith(Flux.fromStream(Streams.stream(scenarioCyclicIterator)))
-        .parallel(Runtime.getRuntime().availableProcessors() * PAR_RATIO)
-        .runOn(Schedulers.elastic())
         .doOnTerminate(this::notifyAwaiting)
-        .subscribe(spec -> Mono.fromFuture(client.executeRequest(
-            getRequestBuilder(
-                spec.getT2(),
-                spec.getT1()),
-            new HttpSpecAsyncHandler(
-                spec.getT1().getUser().getId(),
-                spec.getT2().getEnclosingSpec(),
-                spec.getT2().getStepName(), simulation))
-            .toCompletableFuture()));
+        .flatMap(tuple -> {
+          var session = tuple.getT1();
+          var currentSpec = tuple.getT2();
+
+          var m = Mono.fromFuture(request(client, session, currentSpec).toCompletableFuture());
+          var nextS = currentSpec;
+          while (nextS.getAndThen().isPresent()) {
+            m = m.flatMap(response -> {
+              session.add("response", response);
+              var nextSpec = (HttpSpec) currentSpec.getAndThen().get().apply(session);
+              return Mono.fromFuture(request(client, session, nextSpec).toCompletableFuture());
+            });
+
+            break;
```

Comment:
This needs to be removed in future commits. I am not sure whether the while-loop is the best option here. @luxmeter  any thoughts?

Reply:
I think some kind of loop is necessary. But another thought: Reactor provides already a session mechanism that could be utilised here: https://projectreactor.io/docs/core/release/reference/#context (just overflew it)